src/metabase/lib/binning.cljc

```diff
@@ -194,3 +203,32 @@
           {:bin-width bin-width
            :min-value value
            :max-value (+ value bin-width)})))))
+
+(defn- binning_info->binning-options
+  [binning_info]
+  (-> binning_info
+      u/kebab-keys
+      (set/rename-keys {:binning-strategy :strategy})))
+
+(defn ends-with-binning?
+  "Decide whether string `s` has binning suffix based on `binning-options`."
+  [s binning-options semantic-type]
+  (str/ends-with? s (str ": " (binning-display-name binning-options semantic-type))))
+
+(defn ensure-ends-with-binning
+  "Ensure that `s` is suffixed by appropriate binning info."
+  [s binning-options semantic-type]
+  (if (or (not (string? s))
+          (not (:strategy binning-options))
+          (ends-with-binning? s binning-options semantic-type))
+    s
+    (lib.util/format "%s: %s" s (binning-display-name binning-options semantic-type))))
+
+(defn ensure-binning-in-display-name
+  "Update results column so binning is contained in its display_name."
+  [column]
+  (if (:binning_info column)
```

Comment:
I'm surprised to see this legacy, JS style name in non-JS-facing lib code. We should be converting this in `lib.js` or in `lib.js.metadata`.

Reply:
This function is called on legacy style result metadata column (as mentioned in my previous comment). Reading your comment, I have a feeling this function does not belong to MLv2. The similar approach is taken in my other PR with bucketing. I should move both functions out of lib together in a follow-up. Opened task to track that